packages/dds/tree/src/simple-tree/schemaFactory.ts

```diff
@@ -131,17 +131,27 @@ type UnbrandedName<T extends FlexLeafNodeSchema> = T["name"] extends TreeNodeSch
  *
  * @sealed @public
  */
-export class SchemaFactory<TScope extends string = string, TName extends number | string = string> {
+export class SchemaFactory<
+	TScope extends string | undefined = string,
+	TName extends number | string = string,
```

Comment:
I think most users of "SchemaFactory" would expect any schema factory they have to be assignable to the SchemaFactory type.

As the defaults are setup here, this is not that case. That only works if the defaults allow all possible values of the parameter, and the parameters are covariant.

That said, we have no way for TName to infer to string by default if we do that, and I feel number | string is not what we want instance to infer to by default.

Since we can't actually make everything work how I'd like, it doesn't matter much, but I think:
```suggestion
	out TScope extends string | undefined = string | undefined,
	TName extends number | string = string,
```
better preserves the intention/functionality we had before.

Reply:
Ah, yeah, good point. Probably good to capture assignability to `SchemaFactory` in tests as well. I'll add those.